sources/whippet/src/main.rs

```diff
@@ -0,0 +1,187 @@
+    { allow = true , receive_sender = "*"},   # This is a receive rule
+]
+```
+
+### Policy drop-in files
+
+The original dbus-daemon implmentation allows system services to provide their own custom policy
+configurations. However, system services are allowed to override configurations of the bus besides
+the additional policies.
+
+*whippet* allows additional policy configurations but restricts services from configuring the bus.
+
+## Dbus User
+
+The original dbus-daemon allowed setting the user the daemon should run as. In whippet, the user for
+the bus is configured in the systemd unit file that runs the bus. This allows whippet to only care about
+starting the broker, and keeps its implementation simpler as it doesn't have to implement any sort of
+privilege-dropping mechanism.
+
+## Rule priority calculation
+
+*whippet* implements the priority assignment as close as possible to what the dbus-launcher does. The dbus-launcher
+numbers the rules in a policy as a whole, considering all contexts in the order they appear In these policies, the
+same rules end up with a different index and therefore a different priority:
+
+```xml
+<policy context="default">
+    <allow user="*" />          // RULE 1, NUMBER = 1
+    <deny send_path="*" />      // RULE 2, NUMBER = 2
+</policy>
+
+<policy user="root">
+    <allow send_path="*" />     // RULE 3, NUMBER = 3
+</policy>
+```
+
+```xml
+<policy user="root">
+    <allow send_path="*" />     // WAS RULE 3, NUMBER = 1
+</policy>
+
+<policy context="default">
+    <allow user="*" />          // WAS RULE 1, NUMBER = 2
+    <deny send_path="*" />      // WAS RULE 2, NUMBER = 3
+</policy>
+
+```
+
+In whippet, the order of the contexts doesn't affect the number of the rule. This difference in order doesn’t
+affect the final result of the policy, as the formula to calculate the rules guarantees that user rules will
+always get a higher priority to the default rules. For example, these policies will have the same priorities:
+
+```xml
+<policy user="root">
+    <allow send_path="*" />
+</policy>
+
+<policy context="default">
+    <allow user="*" />
+    <deny send_path="*" />
+</policy>
+
+```
+
+```toml
+[user.root]
+rules = [
+    { allow=true, send_path="*" },
+]
+
+[default]
+rules = [
+    { allow=true, user="*" },
+    { allow=false, send_path="*" }
+]
+
+```
+*/
+
+#[macro_use]
+extern crate log;
+
+mod broker;
+mod child;
+mod dbus_policy;
+mod error;
+mod policy;
+mod socket;
+
+use crate::dbus_policy::DbusPolicy;
+use crate::error::Result;
+use crate::policy::Policy;
+use argh::FromArgs;
+use broker::BrokerManager;
+use log::LevelFilter;
+use simplelog::{Config as LogConfig, SimpleLogger};
+use snafu::ResultExt;
+
+/// D-Bus launcher replacement for Bottlerocket
+#[derive(FromArgs)]
+struct Args {
+    /// path to TOML configuration file
+    #[argh(
+        option,
+        long = "config-file",
+        default = "String::from(\"/usr/share/dbus-1/system.toml\")"
```

Comment:
Let's not put non-standard config files in the standard `/usr/share/dbus-1` path:
```suggestion
        default = "String::from(\"/usr/share/whippet/system.toml\")"
```